eclair-core/src/main/scala/fr/acinq/eclair/payment/Bolt11Invoice.scala

```diff
@@ -86,6 +89,8 @@ case class Bolt11Invoice(prefix: String, amount_opt: Option[MilliSatoshi], creat
 
   lazy val routingInfo: Seq[Seq[ExtraHop]] = tags.collect { case t: RoutingInfo => t.path }
 
+  lazy val extraEdges: Seq[GraphEdge] = routingInfo.flatMap(path => toExtraEdges(path, nodeId))
```

Comment:
As discussed offline, there shouldn't be a dependency on internal router/graph types from `Bolt1Invoice`, which represents a spec object. It should be the other way around, e.g. something like: `GraphEdge.fromHints()`.

```scala
object GraphEdge {
  def fromInvoiceHints(invoice: Bolt11.Invoice): Seq[GraphEdge]
}
```

Btw, I wonder what would be the relationship between this method and the existing `def apply(ac: AssistedChannel): GraphEdge`?

Reply:
Done. And I've removed `AssistedChannel` that was redundant.